### PR TITLE
Add integration tests and ci jobs 

### PR DESCRIPTION
**Pull request by jat-canonical**
This commit adds 
- Github Actions CI jobs to build and test the charm
- Add integration tests to validate relation for the charm

### CODE REVIEW

File: src/interface.py [hunk 1]
```diff
@@ -213,7 +220,11 @@ def _on_relation_changed(self, event: RelationChangedEvent):
         nodes = json.loads(local_data.get("nodes", "[]"))
         current_node = next(node for node in nodes if node["endpoint"] == self.state.endpoint)
 
-        trusted_fps = set(json.loads(current_node["trusted_certs_fp"]))
+        trusted_fps = set(
+            json.loads(current_node["trusted_certs_fp"])
```
Comment (ajanon) [inline src/interface.py:224]: Shouldn't this be .get with a default value, in case the dictionary doesn't have the key? Or is that impossible?

**Review by ajanon — CHANGES_REQUESTED** [review]
One small question.

Also, what's the policy around copyright headers for charms? Is that something you should put in the files you add?

File: src/interface.py (hunk 1 above)
Comment (jat-canonical) [reply src/interface.py:224]: I was asking myself the same question when I saw this. Ideally I would like to put a .get in there but the code down below might get a side effect from the empty value hence decided against it. Also I think this is supposed to break anyway if there is problem with this as it is on a critical code path and this piece of data is written by this code itself.

**Comment by jat-canonical** [comment]
> One small question.
> 
> Also, what's the policy around copyright headers for charms? Is that something you should put in the files you add?

Good question, I am not sure either since these are open repos. @morphis any idea ?

File: src/interface.py (hunk 1 above)
Comment (ajanon) [reply src/interface.py:224]: I see, thanks for the answer!

**Comment by morphis** [comment]
> Good question, I am not sure either since these are open repos. @morphis any idea ?

Yes, we should add them. See [here](https://github.com/canonical/grafana-k8s-operator/blob/main/src/charm.py) for a good example.

**Review by morphis — APPROVED** [review]
LGTM

We can do copyright headers in a separate PR or this one. Will leave this up to you. However we should apply them across all our charms, at best in one go.

**Review by ajanon — APPROVED** [review]
LGTM

**Pull request merged**